tests/suites/test_suite_pk.function

```diff
@@ -101,16 +101,21 @@ static int pk_genkey(mbedtls_pk_context *pk, int parameter)
         mbedtls_pk_get_type(pk) == MBEDTLS_PK_ECKEY_DH ||
         mbedtls_pk_get_type(pk) == MBEDTLS_PK_ECDSA) {
         int ret;
-        if ((ret = mbedtls_ecp_group_load(&mbedtls_pk_ec(*pk)->grp,
-                                          parameter)) != 0) {
+
+        ret = mbedtls_ecp_group_load(&mbedtls_pk_ec(*pk)->grp, parameter);
+        if (ret != 0) {
             return ret;
         }
-
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-        return pk_genkey_ec(&mbedtls_pk_ec(*pk)->grp,
-                            &mbedtls_pk_ec(*pk)->d,
-                            &mbedtls_pk_ec(*pk)->Q);
+        ret = pk_genkey_ec(&mbedtls_pk_ec(*pk)->grp,
+                           &mbedtls_pk_ec(*pk)->d,
+                           &mbedtls_pk_ec(*pk)->Q);
+        if (ret != 0) {
+            return ret;
+        }
+        return mbedtls_pk_update_public_key_from_keypair(pk);
```

Comment:
Perhaps this should be integrated into `pk_genkey_ec()` because I think all callers are going to want to do that.